plugin/teksi_wastewater/utils/database_utils.py

```diff
@@ -220,3 +221,22 @@ def get_validity_check_issues() -> List[str]:
             messages.append("Symbology triggers are disabled")
 
         return messages
+
+    @staticmethod
+    def dblisten(channel: str = None, timeout: float = 5.0):
+        if channel:
+            with DatabaseUtils.PsycopgConnection() as connection:
+                cursor = connection.cursor()
+                cursor.execute(f"LISTEN {channel}")
+                try:
+                    while True:
+                        if select.select([connection], [], [], timeout) == ([], [], []):
+                            print(f"Timeout on Channel {channel}")
+                        else:
+                            connection.poll()
+                            while connection.notifies:
+                                notify = connection.notifies.pop()
+                                yield notify.payload
```

Comment:
This part is blocking the gui thread and makes qgis freeze. It should be moved into a separate thread and communicate with the gui via signal/slots